force-app/main/default/classes/RD2_ChangeHistoryService.cls

```diff
@@ -70,67 +101,70 @@ public without sharing class RD2_ChangeHistoryService {
         SObjectField previousField;
     }
 
-    private Map<SObjectField, ChangeHistoryFieldSet> recurringDonationToChangeHistoryFields = new Map<SObjectField, ChangeHistoryFieldSet>{
-        rdAmountField => new ChangeHistoryFieldSet(RecurringDonationChangeHistory__c.NewAmount__c, 
-            RecurringDonationChangeHistory__c.PreviousAmount__c),
-        rdInstallmentPeriodField => new ChangeHistoryFieldSet(RecurringDonationChangeHistory__c.NewInstallmentPeriod__c, 
-            RecurringDonationChangeHistory__c.PreviousInstallmentPeriod__c),
-        rdInstallmentFrequencyField => new ChangeHistoryFieldSet(RecurringDonationChangeHistory__c.NewInstallmentFrequency__c, 
-            RecurringDonationChangeHistory__c.PreviousInstallmentFrequency__c),
-        rdStatusField => new ChangeHistoryFieldSet(RecurringDonationChangeHistory__c.NewStatus__c, 
-            RecurringDonationChangeHistory__c.PreviousStatus__c),
-        rdStatusReasonField => new ChangeHistoryFieldSet(RecurringDonationChangeHistory__c.NewStatusReason__c, 
-            RecurringDonationChangeHistory__c.PreviousStatusReason__c),
-        rdPaymentMethodField => new ChangeHistoryFieldSet(RecurringDonationChangeHistory__c.NewPaymentMethod__c, 
-            RecurringDonationChangeHistory__c.PreviousPaymentMethod__c),
-        rdCampaignField => new ChangeHistoryFieldSet(RecurringDonationChangeHistory__c.NewCampaign__c, 
-            RecurringDonationChangeHistory__c.PreviousCampaign__c)
-    };
-
-    /**
-     * @description Returns true if any of the key fields used to create Change History has been changed.
-     * @param rd Changed Recurring Donation
-     * @param oldRd Old Recurring Donation
-     * @return Boolean
-     */
-    public void handleHistoryFieldChanged(RD2_RecurringDonation rd, RD2_RecurringDonation oldRd) {
-        RecurringDonationChangeHistory__c rdch;
-
-        for(SObjectField rdField : recurringDonationToChangeHistoryFields.keySet()){
-
-            if(rd.fieldChanged(rdField, oldRd)){
-                if(rdch == null){
-                    rdch = returnChangeHistory(rd, oldRd);
+    private without sharing class ChangeHistoryCreator {
```

Comment:
Nice. My only comment here is to add apexdoc comments to the class and methods - being sure to add something about why this inner class uses `without sharing`.

Reply:
Not usually a proponent of inner classes, but I see why we are using this one and I like that we have a discrete reason for doing so. Agree with @force2b about documentation on the without sharing.